lib/SILOptimizer/Transforms/ARCCodeMotion.cpp

```diff
@@ -375,16 +377,50 @@ bool RetainCodeMotionContext::requireIteration() {
 
 void RetainCodeMotionContext::initializeCodeMotionDataFlow() {
   // Find all the RC roots in the function.
+  llvm::DenseMap<SILValue, Atomicity> AtomicityOfRoot;
```

Comment:
I think you don' t need this map. You can store the info directly in the RCRootAtomicityVault, because you always know the index of the root.